Credify/Credify/serviceX.swift

```diff
@@ -30,14 +34,44 @@ public struct serviceX {
         ///   - from: ViewController that renders a new view from
         ///   - user: User object
         ///   - completion: Completion handler
-        public func showMypage(from: UIViewController, user: CredifyUserModel, completion: @escaping (() -> Void)) {
-            AppState.shared.dismissCompletion = completion
-            let context = PassportContext.mypage(user: user)
-            let vc = WebViewController.instantiate(context: context)
-            let navigationController = UINavigationController(rootViewController: vc)
-            navigationController.modalPresentationStyle = .overFullScreen
-            navigationController.interactivePopGestureRecognizer?.isEnabled = false // disable navigation bar swipe back
-            from.present(navigationController, animated: true)
+        public func showMypage(
+            from: UIViewController,
+            user: CredifyUserModel,
+            pushClaimTokensTask: @escaping ((String, ((Bool) -> Void)?) -> Void),
+            completion: @escaping (() -> Void)
+        ) {
+            let tableName = "serviceX"
+            var errorMessage = ""
+            
+            // Phone number
+            let countryCode = user.countryCode.trimmingCharacters(in: .whitespacesAndNewlines)
+            let phoneNumber = user.phoneNumber.trimmingCharacters(in: .whitespacesAndNewlines)
+            if countryCode.isEmpty || phoneNumber.isEmpty || phoneNumber.count < 8 || phoneNumber.count > 12 {
+                errorMessage.append(String(format:"FieldIsInvalid".localized(tableName: tableName), "\'Phone number\'"))
+                errorMessage.append(" ")
+            }
```

Comment:
This file should be as simple as possible. You should move this validation to another file

Reply:
I moved to ValidationUtils file